inc/saitypes.h

```diff
@@ -1196,6 +1199,94 @@ typedef struct _sai_port_snr_list_t
     sai_port_snr_values_t *list;
 } sai_port_snr_list_t;
 
+/**
+ * @brief poe port active channel (when delivering power)
+ */
+typedef enum _sai_poe_port_active_channel_type_t
+{
+    SAI_POE_PORT_ACTIVE_CHANNEL_TYPE_A = 0,
+    SAI_POE_PORT_ACTIVE_CHANNEL_TYPE_B = 1,
+    SAI_POE_PORT_ACTIVE_CHANNEL_TYPE_A_AND_B = 2
+} sai_poe_port_active_channel_type_t;
+
+/**
+ * @brief poe port signature type (when delivering power) 
+ */
+typedef enum _sai_poe_port_signature_type_t
+{
+    SAI_POE_PORT_SIGNATURE_TYPE_SINGLE = 0,
+    SAI_POE_PORT_SIGNATURE_TYPE_DUAL = 1
+} sai_poe_port_signature_type_t;
+
+/**
+ * @brief poe port classification method (when delivering power)
+ */
+typedef enum _sai_poe_port_class_method_type_t
+{
+    SAI_POE_PORT_CLASS_METHOD_TYPE_REGULAR = 0,
+    SAI_POE_PORT_CLASS_METHOD_TYPE_AUTO_CLASS = 1
+} sai_poe_port_class_method_type_t;
+
+/**
+ * @brief Defines a port consumption structure
+ * 
+ * Data that is needed and available when a port is delivering power
+ * 
+ */
+typedef struct _sai_poe_port_power_consumption_t 
+{
+    /**
+     * @brief active channel: a/b/ab
+     */
+    sai_poe_port_active_channel_type_t active_channel;
+
+    /**
+     * @brief voltage in milliVolts
+     */
+    uint32_t voltage_in_mv;
+
+    /**
+     * @brief current in milliAmpere
+     */
+    uint32_t current_in_ma;
+
+    /**
+     * @brief consumption in milliWatts
+     */
+    uint32_t consumption_in_mw;
+
```

Comment:
please remove units from vairable names

Reply:
also is power consumption not needed information ? you can calculate this multiplying A and V to het Watts, but maybe there is some other meaning of this ?
also not sure what are the expected ranges between voltage and current are 40-50 volts ?

Reply:
I'll remove them. :)

Theoretically yes, you can calculate the Watts. However, we need to also take into consideration the power loss that happens because of the cable length and other hardware components/designs. So, when we multiple A and V we get a number that is in the ballpark but not completely accurate. This is why we prefer to add a dedicated consumption field.

As for the second question, the expected ranges are based on the PoE class. Each class supports up to a certain number of watts. Here is a chart that has the ranges:
![image](https://github.com/opencomputeproject/SAI/assets/43090733/22b824ed-6889-4825-8fe9-99dc908fb626)

Reply:
Yes. That's correct. We need to account the power conversion factors.
Power = current * voltage * power_factor
Here the power factor could be 80% to 85%.
The PSE reports the power at it's side.  It means the power that the PSE is delivering. This doesn't include the power loss due to cable medium and length. The loss is accounted at PD side.
So please keep the power value fetched from the PoE controller that is accurate.

Reply:
I am keeping the power value/consumption field. :)